src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -558,6 +558,13 @@ protected Query doToQuery(QueryShardContext context) {
             radius = knnEngine.scoreToRadialThreshold(this.minScore, spaceType);
         }
 
+        if (radius != null
+            && methodParameters != null
+            && methodParameters.containsKey(METHOD_PARAMETER_EF_SEARCH)
+            && knnEngine.equals(KNNEngine.LUCENE)) {
+            throw new IllegalArgumentException(String.format("[" + NAME + "] Lucene engine does not support ef_search with radial search"));
+        }
```

Comment:
This is engine specific, we need to move it to EngineSpecificMethodContext. Currently the `supportedMethodParameters()` does not support this. We need to modify the interface to embed this logic in there. Can we change `supportedMethodParameters()` to add a context parameter where we can pass in the information and decide at run time which method parameters to return. Something like this

```
class LuceneHNSWContext implements EngineSpecificMethodContext {

Map<String, ?> supportedMethodParameters(Context ctx) {
HNSWContext ctx = (HNSWContext) ctx;
if(ctx.isRadial()) {
return emptyMap()
}

return supportedMethodParams
}

}
```

Reply:
Good point, let me check.

Reply:
Updated